test/e2e/framework/service/wait.go

```diff
@@ -63,3 +67,49 @@ func WaitForServiceResponding(c clientset.Interface, ns, name string) error {
 		return true, nil
 	})
 }
+
+func WaitForServiceDeletedWithFinalizer(cs clientset.Interface, namespace, name string) {
```

Comment:
These two functions are moved without modification to allow reusing --- the only change is replacing the hardcoded "service.kubernetes.io/load-balancer-cleanup" with the imported one.